src/DiagnosticManager.ts

```diff
@@ -234,7 +234,7 @@ export class DiagnosticManager {
                     isMatch = !!context.tags?.includes(filter.tag);
                 }
                 if (isMatch && needToMatch.scope) {
-                    isMatch = context.scope === filter.scope;
+                    isMatch = context.scope?.name === filter.scope.name;
```

Comment:
if an xml file changes, then there is a new scope created, so a check on `context.scope === filter.scope` wouldn't work to remove old diagnostics